src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+    def experiment_name(self, a_name: str) -> None:
+        """
+        User can set the experiment that they want to load.
+        """
+        if a_name != self._experiment_name:
+            # Added this to prevent errors with incorrect file name
+            if a_name in self.light.experiment.GetSavedExperiments(): 
+                self._experiment_name = a_name
+                self.light.load_experiment(self._experiment_name)
+            else:
+                logger.error(f"An experiment with that file name does not exist.")
+
+    @property
+    def grating(self) -> str:
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, value: str) -> None:
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        if value in self.grating_options:
+            self.light.set(lf.AddIns.SpectrometerSettings.GratingSelected, String(value))
+        else:
+            logger.error(f"Grating {value} is not an options. The options are: {self.grating_options}")
+
+    @property
+    def grating_options(self) -> List[str]:
+        """
+        Returns a list of all installed gratings.
+        """
+        # This line below is critical. "GetCurrentCapabilities" is able to return the list of possibilities of any Lightfield call in order to provide more information.
+        available_gratings = self.light.experiment.GetCurrentCapabilities(lf.AddIns.SpectrometerSettings.GratingSelected)
+        return [str(a) for a in available_gratings]
+
+    @property
+    def num_frames(self) -> int:
+        """
+        Returns the number of frames taken during the acquisition.
+        """
+        return self.light.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
+
+    @num_frames.setter
+    def num_frames(self, num_frames: int) -> None:
+        """
+        Sets the number of frames to be taken during acquisition to number.
+        """
+        self.light.set(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore, Int64(num_frames))
+
+    @property
+    def exposure_time(self) -> float:
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms: float) -> None:
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, Double(ms))
+
+    @property
+    def temperature_sensor_setpoint(self) -> float:
+        """
+        Returns the sensor setpoint temperature (Celsius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+
+    @temperature_sensor_setpoint.setter
+    def temperature_sensor_setpoint(self, deg_C: float) -> None:
```

Comment:
rename to temperature_setpoint

Reply:
following the idea of matching light field, I would actually recommend `sensor_temperature_set_point`.  Make it exactly like the setting name in the documentation.